lib/HomeyCompose.js

```diff
@@ -263,18 +263,16 @@ class HomeyCompose {
               const templateCards = flowTemplates[type];
 
               let flowTemplate = {};
-              let templateId;
-              for (const k of Object.keys(templateIds)) {
-                templateId = templateIds[k];
-                if (!templateCards.templateId) {
+              for (const templateId of templateIds) {
+                if (!templateCards[templateId]) {
                   throw new Error(`Invalid driver flow template for driver ${driverId}: ${templateId}`);
                 }
                 flowTemplate = Object.assign(flowTemplate, templateCards[templateId]);
               }
 
               // assign template to original flow object
               Object.assign(card, {
-                id: card.$id || templateId,
+                id: card.$id || templateIds[templateIds.length - 1],
```

Comment:
I have no idea why we do this... Pretty confused in general as to where the `$id` properties come from.